src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -171,4 +171,66 @@ sealed class BitPat(val value: BigInt, val mask: BigInt, width: Int) extends Sou
   }.mkString
 
   override def toString = s"BitPat($rawString)"
+
+  /**
+    * If a BitPat overlaps with another BitPat, i.e., exists x : UInt that (x === this) && (x === rhs)
+    *
+    * (lhs overlap rhs) iff bitwise: both care (lhs.mask & rhs.mask) => both equal (lhs.value == rhs.value)
+    */
+  def overlap(rhs: BitPat): Boolean = ((mask & rhs.mask) & (value ^ rhs.value)) == 0
+  /**
+    * If a BitPat contains another BitPat
+    *
+    * (lhs contain rhs) iff bitwise: lhs care => rhs also care && equal (~(rhs.mask & (value xnor rhs.value)))
+    */
+  def contain(rhs: BitPat): Boolean = (mask & (~rhs.mask | (value ^ rhs.value))) == 0
```

Comment:
```suggestion
  def supersetOf(rhs: BitPat): Boolean = (mask & (~rhs.mask | (value ^ rhs.value))) == 0
```
Scala naming would suggest calling this `contains`, but that operations is more for containing an element rather than checking the relationship between two `Sets`. Scala Set does define `subsetOf`, and this seems to be similar but `supersetOf` so I'd suggest following that sort of naming convention.

(Currently, `===` is more similar with Scala's `.contains` method, perhaps we should add `.contains` as an alias for `===`)